### ISSUE TITLE
Intercom::ResourceNotFound:scroll parameter not found

### ISSUE THREAD

**Issue by andrewhavens** [intercom/intercom-ruby#450]
While iterating over contacts scroll list, it runs into an error after reaching 100th contact in list. After reading through the readme docs on using scroll, I think the scroll param might be timing out by taking more than 60 seconds before it reaches item 100, but the docs aren't clear on how to fix this (or request a new scroll param).

## Version info
  - intercom-ruby version: 3.7.1
  - Ruby version: 2.5.1

## Expected behavior
It continues iterating without issue

## Actual behavior
It results in an error.

## Steps to reproduce
```ruby
start_time = Time.now.to_i
begin
  puts "Starting..."
  $intercom.contacts.scroll.each do |contact|
    # make some more API requests
  end
rescue => e
  puts e.inspect
end
puts "Total time #{Time.now.to_i - start_time} seconds."
```

## Logs
```
Starting...
Error: #Intercom::ResourceNotFound:scroll parameter not found @http_code=404 @application_error_code="not_found" @field=nil @request_id="002qm6o5t66bvlqrcbt0"
Total time 224 seconds.
```

**Comment by theandrewykim** [comment]
@andrewhavens  Hi Andrew! Great name by the way.
 
Just to be clear in your script - you mentioned in your commented code
`# make some more API requests`

Are you making requests with the data being returned as you iterate through contacts?
I'd recommend making your other API requests separately - perhaps you could push the contacts you wanted to do something with to a separate data structure?

By just iterating through users, I was able to iterate through a list of 1000 users in under 25 seconds.

**Comment by andrewhavens** [comment]
@theandrewykim Right, that comment is pseudo-code for some slow API requests, causing the total time for iterating over 100 to take longer than 60 seconds (224 seconds). I was able to get around this issue by copying the contact information to a CSV file, then processing the list from the CSV file. This works, but was an unexpected workaround. It would be nice if the error/solution was a little more clear/user-friendly.

**Comment by andrewhavens** [comment]
Thanks! To be clear, I was working with a total of around 60,000 contacts and chose to store the data in a CSV file in order to avoid memory bloat, and to resume the list in case an error caused my script to crash.

**Comment by theandrewykim** [comment]
@andrewhavens Noted! Thanks for letting me know. Definitely puts your request into perspective with that context :)
